lib/alg/bellman-ford.js

```diff
@@ -0,0 +1,74 @@
+var _ = require("../lodash");
+
+module.exports = bellmanFord;
+
+var DEFAULT_WEIGHT_FUNC = _.constant(1);
+
+function bellmanFord(g, source, weightFn, edgeFn) {
+  return runBellmanFord(g, String(source),
+                     weightFn || DEFAULT_WEIGHT_FUNC,
+                     edgeFn || function(v) { return g.outEdges(v); });
+}
+
+
+function runBellmanFord(g, source, weightFn, edgeFn) {
+  var results = {},
+      didADistanceUpgrade = true,
+      iterations = 0,
+      nodes = g.nodes(),
+      edges = [];
+
+  // Get the edges using the edgeFn provided
+  nodes.forEach(function(v){
+    edgeFn(v).forEach(function(e){
+      if(typeof(e)==="object"){
```

Comment:
Yes, the edge functions should always return an array of edge objects, but inEdges() for example, which is also used on a dijkstra test, returns:
- an empty object if the node has no edges
- an edge object if the node is associated with only one edge 
- an array of edges otherwise

Reply:
I stand corrected, edge functions do always return an array of objects. The check here is unnecessary.